deliver/lib/deliver/upload_metadata.rb

```diff
@@ -118,8 +118,14 @@ def upload(options)
 
       UI.message("Uploading metadata to iTunes Connect")
       v.save!
-      details.save!
-      UI.success("Successfully uploaded set of metadata to iTunes Connect")
+      begin
+        details.save!
+        UI.success("Successfully uploaded set of metadata to iTunes Connect")
+      rescue Spaceship::ITunesConnectError => e
+        if e.message.include?('App Name cannot be longer than 50 characters') || e.message.include?('The app name you entered is already being used')
+          UI.user_error!(e.message)
+        end
```

Comment:
Could you provide some more context for this change? In particular, why do we not re-raise the exception if the error message isn't one of the 2 messages we check for?